presto-main/src/test/java/com/facebook/presto/execution/TestCreateTableTask.java

```diff
@@ -0,0 +1,136 @@
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestCreateTableTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private final ExecutorService executor = newCachedThreadPool(daemonThreadsNamed("test-%s"));
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testCreateTableDoesNotThrowAlreadyExistsExceptionWhenIfExists()
+            throws Exception
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        TypeManager typeManager = new TypeRegistry();
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Catalog bogusTestingCatalog = createBogusTestingCatalog(CATALOG_NAME);
+        AtomicReference<Boolean> metadataCreateTableCalled = new AtomicReference<>(Boolean.FALSE);
+
+        MetadataManager metadata = new MetadataManager(
+                new FeaturesConfig(),
+                typeManager,
+                new BlockEncodingManager(typeManager),
+                new SessionPropertyManager(),
+                new SchemaPropertyManager(),
+                new TablePropertyManager(),
+                transactionManager) {
+            @Override
+            public Optional<TableHandle> getTableHandle(Session session, QualifiedObjectName tableName)
+            {
+                return Optional.empty();
+            }
+            @Override
+            public void createTable(Session session, String catalogName, ConnectorTableMetadata tableMetadata)
+            {
+                metadataCreateTableCalled.set(Boolean.TRUE);
+                throw new PrestoException(ALREADY_EXISTS, "Test that we ignore ALREADY_EXISTS exceptions when notExists set");
+            }
+        };
+
+        AccessControl accessControl = new AccessControlManager(transactionManager);
+
+        metadata.getTablePropertyManager().addProperties(bogusTestingCatalog.getConnectorId(), ImmutableList.of(stringSessionProperty(
+                "baz",
+                "test property",
+                null,
+                false)));
+        catalogManager.registerCatalog(bogusTestingCatalog);
+
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition("a", "BIGINT", Optional.empty())),
+                true,
+                ImmutableMap.of(),
+                Optional.empty());
+
+        Session session = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(TINY_SCHEMA_NAME)
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .build();
+
+        QueryStateMachine stateMachine = QueryStateMachine.begin(new QueryId("create_table"), "CREATE TABLE", session, URI.create("fake://uri"), true, transactionManager, accessControl, executor, metadata);
```

Comment:
The only thing we need from `QueryStateMachine` is a session. `TransactionManager` isn't used at all. We could refactor `CreateTableTask` to have a package-private `@VisibleForTesting` method for execute that takes only the required objects.